rfcs/web_extensions.md

```diff
@@ -0,0 +1,56 @@
+# RFC #219: Add support for WebExtensions in WPT
+
+## Summary
+
+The WebExtensions API extends the capabilities of the browser. Adding support
+for WebExtensions in the web platform tests will increase interoperability
+and will help drive the standardization of this API.
+
+This RFC proposes adding a new `testharness.js` test type, `.extension.js`, to handle
+testing this API, in addition to using `testdriver.js` to load and unload extensions.
+   
+##  Details
+  
+Using `testdriver.js`, we added support for testing the WebExtensions API by loading
+a web extension designed to test the functionality of a specific API.
+The extension will be loaded after the tests begins, and unloaded before the
+test is finished.
+
+Most of the test execution is handled within the extension, via the 
+[browser.test](https://chromium.googlesource.com/chromium/src.git/+/master/extensions/docs/testing_api.md)
+API. We’ve elected to use these APIs since all participating browser vendors use
+`browser.test`  internally and they can easily port over existing tests to
+the web platform tests.
+
+Because these tests won’t leverage `testharness.js` directly, we’ve introduced a new
+`testharness.js`, `.extension.js`, that will create the necessary boilerplate to
+convert the `browser.test` assertions into the corresponding assertions in the test
+harness.
+   
+A proposed patch is available at https://github.com/web-platform-tests/wpt/pull/50648.
+
+## Alternatives considered
+
+We considered loading the extension statically before each `testharness.js` test is run,
+but we decided against that since `testdriver.js` will allow the tests to drive the
+browser, and in the future, test functionality such as opening popups and clicking
+menu items. 
+
+We considered adding a new test type rather than using JavaScript tests, but we decided
+against it because it was a lot more work compared to using `testharness.js`.
+   
+## Risks
+
+There are two potential concerns with this implementation:
+
+1. We have no precedent for tests run via a Classic command in some user agents
```

Comment:
I have concerns that we will end up with two methods for dealing with extensions exposed to test authors following different (and separately evolving) specs that are not fully implemented by all vendors. I think it would be preferable to start with adding methods to testdriver.js that match WebDriver BiDi and allow synchronous handling of these methods for non-WebDriver-BiDi implementations (prototype: https://github.com/web-platform-tests/wpt/pull/51090).

Reply:
We ended up discussing this in the Browser Testing and Tools WG last Tuesday, and it was decided that using the classic approach for `testdriver.js` was ok, and Chrome can continue to use the BiDi implementation in the backend

Reply:
I believe it was the `wpt RFCs and infrastructure` and not the Browser Testing and Tools WG meeting. Could you please add details to the RFC on how using the proposed testdriver.js API would be possible for implementations relying on WebDriver BiDi?